### DISCUSSION TITLE
Is it possible to set global/module wide xfail behavior.

### DISCUSSION THREAD

**Discussion by HenrikSpiegel** [pytest-dev/pytest#11474] [Q&A]
When working in a test-first approach it would be great to be able to mark all tests in a given document.

ie. catch all test which raises NotImplementedError and mark them for xfail.

Currently I am doing this for every test and it would be great to be able to set a default behavior.

Is there a better way than to mark each test?

**Comment by RonnyPfannschmidt** [comment]
```python
import pytest

pytestmark = [pytest.mark.xfail(strict=False)]

def test_yay():
  walk.will.fail
  ```
